src/noise.ts

```diff
@@ -111,26 +111,38 @@ export class Noise implements INoiseConnection {
    */
   private async performHandshake(params: HandshakeParams): Promise<IHandshake> {
     const payload = await getPayload(params.localPeer, this.staticKeys.publicKey, this.earlyData);
-
-    const remoteStaticKey = KeyCache.load(params.remotePeer);
+    let tryIK = this.useNoisePipes;
+    if(params.isInitiator && KeyCache.load(params.remotePeer) === null) {
```

Comment:
What should happen if responder doesn't have the remote static key in cache? One test fails because of this condition.

Reply:
responder doesn't need to know initiator remote static key, he decrypts with his private static key as message is encrypted with his public static key. Spec:

Reply:
```
Later, Alice can open a new Noise connection to Bob using the IK pattern. This is a zero-RTT handshake that uses the cached static key to encrypt the initial handshake message.

If Alice attempts an IK handshake but Bob has changed his static Noise key, Bob will fail to decrypt the handshake message.
```

Reply:
Oh, yeah, indeed. Test is wrong :)